feature/platform/transceiver/otg_test/zr_temperature_test/zr_temperature_test.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zr_temperature_test
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"github.com/openconfig/ygnmi/ygnmi"
+	"github.com/openconfig/ygot/ygot"
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// Topology:
+//   dut1:port1 <--> port1:dut2
+//
+
+func TestZRTemperatureState(t *testing.T) {
+	dut1 := ondatra.DUT(t, "dut1")
+	dp1 := dut1.Port(t, "port1")
+	t.Logf("dut1: %v", dut1)
+	t.Logf("dut1 dp1 name: %v", dp1.Name())
+	// Check if TRANSCEIVER is of type 400ZR
+	if dp1.PMD() != ondatra.PMD400GBASEZR {
+		t.Fatalf("%s Transceiver is not 400ZR its of type: %v", dp1, dp1.PMD())
+	}
+	component1 := gnmi.OC().Component(dp1.Name())
+	temperatureInstantdp1 := gnmi.Get(t, dut1, component1.Temperature().Instant().State())
```

Comment:
temperature can be reported in one of the two ways in components tree:
1) in the transceiver hierarchy, or
2) the transceiver may have a temperature sensor subcomponent that reports the values (instant/min/max/avg/..)
e.g. 
https://github.com/openconfig/featureprofiles/blob/dd9eed3fb05f68ed1ae7aa0ce0fe027c2d638cf2/feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go#L106-L115

Reply:
Hi Sergey,

Updated the script as suggested please review and let me know any further changes required.

Thanks